addons/product_expiry/models/production_lot.py

```diff
@@ -15,6 +15,13 @@ class StockProductionLot(models.Model):
         help='This is the date on which the goods with this Serial Number should be removed from the stock.')
     alert_date = fields.Datetime(string='Alert Date',
         help="This is the date on which an alert should be notified about the goods with this Serial Number.")
+    product_expiry_alert = fields.Boolean(compute='_compute_product_expiry_alert')
+
+    @api.depends('alert_date', 'product_id')
+    def _compute_product_expiry_alert(self):
+        current_date = fields.Date.from_string(fields.Date.today())
+        for lot in self.filtered(lambda l: l.alert_date and l.product_id):
```

Comment:
Why do we depend on and check for a product_id here ? Product_id is a required field anyway.

Is there a use case for changing the product of a lot once it's made and why would it change the product_expiry_alert if the alert_date isn't changed ?